src/parpeoptimization/minibatchOptimization.cpp

```diff
@@ -195,11 +200,55 @@ void ParameterUpdaterRmsProp::clearCache() {
     std::fill(oldGradientNormCache.begin(), oldGradientNormCache.end(), 0.0);
 }
 
+void ParameterUpdaterMomentum::initialize(unsigned int numParameters) {
+    momentum.resize(numParameters);
+    oldMomentum.resize(numParameters);
+    std::fill(momentum.begin(), momentum.end(), 0.0);
+    std::fill(oldMomentum.begin(), oldMomentum.end(), 0.0);
+}
+
+void ParameterUpdaterMomentum::updateParameters(double learningRate,
+                                               int  /*iteration*/,
+                                               gsl::span<double const> gradient,
+                                               gsl::span<double> parameters,
+                                               gsl::span<const double> lowerBounds,
+                                               gsl::span<const double> upperBounds) {
+
+    int numParameters = gradient.size();
+    double momentumNormalize = 1.0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'momentumNormalize' can be reduced.](https://app.codacy.com/app/dweindl/parPE/pullRequest?prid=4535376)